waspc/data/Generator/templates/server/src/crud/_operations.ts

```diff
@@ -0,0 +1,215 @@
+{{={= =}=}}
+import prisma from "../dbClient.js";
+
+import type {
+  {=# isAuthEnabled =}
+  AuthenticatedAction,
+  AuthenticatedQuery,
+  {=/ isAuthEnabled =}
+  {=^ isAuthEnabled =}
+  Action,
+  Query,
+  {=/ isAuthEnabled =}
+  _{= crud.entityUpper =},
+} from "../_types";
+import type {
+  Prisma,
+} from "@prisma/client";
+import type {
+  {= crud.entityUpper =},
+} from "../entities";
+{=# isAuthEnabled =}
+import HttpError from "../core/HttpError.js";
+{=/ isAuthEnabled =}
+{=# overrides.GetAll.isDefined =}
+{=& overrides.GetAll.importStatement =}
+{=/ overrides.GetAll.isDefined =}
+{=# overrides.Get.isDefined =}
+{=& overrides.Get.importStatement =}
+{=/ overrides.Get.isDefined =}
+{=# overrides.Create.isDefined =}
+{=& overrides.Create.importStatement =}
+{=/ overrides.Create.isDefined =}
+{=# overrides.Update.isDefined =}
+{=& overrides.Update.importStatement =}
+{=/ overrides.Update.isDefined =}
+{=# overrides.Delete.isDefined =}
+{=& overrides.Delete.importStatement =}
+{=/ overrides.Delete.isDefined =}
+
+type _WaspEntityTagged = _{= crud.entityUpper =}
+type _WaspEntity = {= crud.entityUpper =}
+const entities = {
+  {= crud.entityUpper =}: prisma.{= crud.entityLower =},
+}
+
+{=# crud.operations.GetAll =}
+{=# isAuthEnabled =}
+export type GetAllQuery<Input, Output> = AuthenticatedQuery<[_WaspEntityTagged], Input, Output>
+{=/ isAuthEnabled =}
+{=^ isAuthEnabled =}
+export type GetAllQuery<Input, Output> = Query<[_WaspEntityTagged], Input, Output>
+{=/ isAuthEnabled =}
+{=# overrides.GetAll.isDefined =}
+const _waspGetAllQuery = {= overrides.GetAll.importIdentifier =}
+{=/ overrides.GetAll.isDefined =}
+{=^ overrides.GetAll.isDefined =}
+type GetAllInput = {}
+type GetAllOutput = _WaspEntity[]
+const _waspGetAllQuery: GetAllQuery<GetAllInput, GetAllOutput> = ((args, context) => {
+  {=^ crud.operations.GetAll.isPublic =}
+  throwIfNotAuthenticated(context)
+  {=/ crud.operations.GetAll.isPublic =}
+  return context.entities.{= crud.entityUpper =}.findMany();
+});
+{=/ overrides.GetAll.isDefined =}
+
+export type GetAllQueryResolved = typeof _waspGetAllQuery
```

Comment:
This is the type we import on the client, it's either:
- pointing to our default implementation type
- or custom user implementation
by using `typeof`